### DISCUSSION TITLE
How to make sequence, string and map to bounded?

### DISCUSSION THREAD

**Discussion by wjhtinger** [eProsima/Fast-DDS#3920] [Q&A] [answered]
I want to use these types in Data-sharing. How to limit these data to bounded?

**Comment by RowdenRichardConway** [comment, marked as answer]
I believe in the IDL you can bound things with angled brackets, for instance:

```
struct Foo {
    string<12> myString;
    sequence <short, 10> listOfNumbers;
};
```


should produce a Foo type with a property called myString which can be a maximum of 12 characters in length, and a list/array of 16-bit integers which can contain a maximum of 10 elements.
